### PR TITLE
Add some checks for dirty and formating to ci

### PR DESCRIPTION
**Pull request by tmjd**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -303,6 +305,22 @@ static-checks:
...
 
+.PHONY: format-check
+format-check:
+	@$(CONTAINERIZED) \
+	sh -c '$(GIT_CONFIG_SSH) \
+	files=$$(gofmt -l ./pkg ./controllers ./api ./test); \
+	[ "$$files" = "" ] && exit 0; \
+	echo The following files need a format update:; \
+	echo $$files; \
+	echo Try running \"make fmt\" and committing any changes; \
```
Comment (doublek) [inline Makefile:316]: I think in other repos we prefer using goimports. Could we use that instead?

Comment (tmjd) [reply Makefile:316]: Those aren't the same things it seems. I tried running goimports on the operator and it changed several imports and they looked incorrect.

Comment (doublek) [reply Makefile:316]: Perhaps a difference in go version between containerized and a local non-containerized builds?

Comment (doublek) [reply Makefile:316]: To clarify, I did expect that goimports would do the work of `go fmt` as well as clean up imports. However if the imports look wrong, then my above comment could be one reason for this behavior.

Comment (tmjd) [reply Makefile:316]: Ok yeah looks like if I containerize and use go-build to run goimports (which already exists under the `fix` make target) it doesn't change a bunch of stuff incorrectly.

Comment (tmjd) [reply Makefile:316]: Is it ok using the target we already had that is called `fix` which runs goimport?

Comment (doublek) [reply Makefile]: Awesome. I think that works.

**Review by doublek — APPROVED** [review]
LGTM

**Pull request merged**